gatsby-node.js

```diff
@@ -289,18 +289,57 @@ async function getIndexToUse({ index, tempIndex, enablePartialUpdates }) {
   return index;
 }
 
-async function getSettingsToApply({ settings, index, tempIndex, indexToUse }) {
-  const requestedSettings = settings ? settings : await index.getSettings();
+
+async function getSettingsToApply({
+  settings: givenSettings,
+  index,
+  tempIndex,
+  indexToUse,
+}) {
+  const { replicaUpdateMode, ...settings } = givenSettings;
+  var existingSettings = {};
+  try {
+    existingSettings = await index.getSettings();
+  } catch (e) {
+    report.error(`${e.toString()} ${index.indexName}`);
```

Comment:
I think this should be a fatal error, since it might wipe out the settings in an index